api/handlers/vouchers.go

```diff
@@ -89,6 +94,89 @@ func GetVoucherByGUIDHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// VerifyOwnershipVoucher performs verification of an ownership voucher as per FDO spec section 3.4.6.
+// This function verifies:
+//   - Ownership: Voucher owner key matches one of the server's configured keys
+//   - Header fields: Protocol version, GUID, DeviceInfo, ManufacturerKey, RvInfo, cert chain consistency
+//   - Cryptographic integrity: Signature chain, device cert chain, manufacturer cert chain, cert chain hash
+//
+// Note: The following validations can be performed by the device during TO2 not by owner-server,
+// so not included in this verification function.
+//   - HMAC verification (ov.VerifyHeader): Owner server does not have the device HMAC secret;
+//     this verification is performed by the device during TO2 protocol
+//   - Manufacturer key hash verification (ov.VerifyManufacturerKey): Requires trusted manufacturer
+//     key hashes to be configured (owner server has no source for these hashes); this verification
+//     is performed by the device during TO2 protocol using the manufacturer key hash stored in the
+//     device credential from DI (Device Initialization)
+
+func VerifyOwnershipVoucher(ov *fdo.Voucher, ownerPKeys []crypto.PublicKey) error {
+	if len(ownerPKeys) == 0 {
+		return fmt.Errorf("ownerPKeys must contain at least one owner public key")
+	}
+
+	// Ownership Verification (fail fast if not our voucher)
+	expectedPubKey, err := ov.OwnerPublicKey()
+	if err != nil {
+		return fmt.Errorf("unable to parse owner public key from voucher: %w", err)
+	}
+
+	type equalable interface{ Equal(crypto.PublicKey) bool }
+
+	var ownerKeyMatched bool
+	for _, possibleOwnerKey := range ownerPKeys {
+		eq, ok := possibleOwnerKey.(equalable)
+		if !ok {
+			return fmt.Errorf("owner key type does not support equality comparison")
+		}
+		if eq.Equal(expectedPubKey) {
+			ownerKeyMatched = true
+			break
+		}
+	}
+	if !ownerKeyMatched {
+		return fmt.Errorf("voucher owner key does not match any of the server's configured keys")
+	}
```

Comment:
Let's simplify this a bit and comment not obvious things, the cast is not supposed to fail because `crypto.PublicKey` implements the interface (you will need to import `slices` btw):

```suggestion
	// Cast is needed to call Equal()
	// See: https://pkg.go.dev/crypto#PublicKey
	if !slices.ContainsFunc(ownerPKeys, expectedPubKey.(interface{ Equal(crypto.PublicKey) bool }).Equal) {
		return fmt.Errorf("voucher owner key does not match any of the server's configured keys")
	}
```